cluster/points_writer.go

```diff
@@ -196,6 +202,13 @@ func (w *PointsWriter) WritePoints(database, retentionPolicy string, consistency
 	w.statMap.Add(statWriteReq, 1)
 	w.statMap.Add(statPointWriteReq, int64(len(points)))
 
+	if w.MaxConcurrentWrites > 0 {
```

Comment:
It may be best to only include the http connection limit then since I don't think a write limit would solve the underlying issue for #6559, while the generic max connection limit solves that problem perfectly.

Do we still want the `/write`, `/query`, and generic http handlers to have separate connection limits?

I'll start working on a change to address the other feedback. It seems like we should not rely on the `max-concurrent-queries` option to address this issue and `/query` should have its own connection limit if we're going to do that for the others too. It may also be better to try implementing this at the socket level, but I don't think Go provides the ability to perform an action when an HTTP connection starts/stops.